src/Atmos/Parameterizations/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -45,146 +42,165 @@ module SurfaceFluxes
-  return (grav * alpha0_0 / cp_ / T_b * temp2)
-end
-
-module Byun1990
-
-using ...RootSolvers
-using ...MoistThermodynamics
-using ...PlanetParameters
+    SurfaceFluxConditions{T}
 
-""" Computes ψ_m for stable case. See Eq. 12 Ref. Byun1990 """
-ψ_m_stable(ζ, ζ_0, β_m) = -β_m * (ζ - ζ_0)
+Surface flux conditions, returned from `surface_conditions`.
 
-""" Computes ψ_h for stable case. See Eq. 13 Ref. Byun1990 """
-ψ_h_stable(ζ, ζ_0, β_h) = -β_h * (ζ - ζ_0)
+# Fields
 
-""" Computes ψ_m for unstable case. See Eq. 14 Ref. Byun1990 """
-function ψ_m_unstable(ζ, ζ_0, γ_m)
-  x(ζ) = sqrt(sqrt(1 - γ_m * ζ))
-  temp(ζ, ζ_0) = log1p((ζ - ζ_0)/(1 + ζ_0))  # log((1 + ζ)/(1 + ζ_0))
-  ψ_m = (2 * temp(x(ζ), x(ζ_0)) + temp(x(ζ)^2, x(ζ_0)^2) - 2 * atan(x(ζ)) + 2 * atan(x(ζ_0)))
-  return ψ_m
+$(DocStringExtensions.FIELDS)
+"""
+struct SurfaceFluxConditions{T}
+  momentum_flux::T
+  buoyancy_flux::T
+  Monin_Obukhov_length::T
+  friction_velocity::T
+  temperature_scale::T
+  pottemp_flux::T
+  exchange_coeff_momentum::T
+  exchange_coeff_heat::T
 end
 
-""" Computes ψ_h for unstable case. See Eq. 15 Ref. Byun1990 """
-function ψ_h_unstable(ζ, ζ_0, γ_h)
-  y(ζ) = sqrt(1 - γ_h * ζ )
-  ψ_h = 2 * log1p((y(ζ) - y(ζ_0))/(1 + y(ζ_0))) # log((1 + y(ζ))/(1 + y(ζ_0)))
-  return ψ_h
+function Base.show(io::IO, sfc::SurfaceFluxConditions)
+  println(io, "----------------------- SurfaceFluxConditions")
+  println(io, "momentum_flux           = ", sfc.momentum_flux)
+  println(io, "buoyancy_flux           = ", sfc.buoyancy_flux)
+  println(io, "Monin_Obukhov_length    = ", sfc.Monin_Obukhov_length)
+  println(io, "friction_velocity       = ", sfc.friction_velocity)
+  println(io, "temperature_scale       = ", sfc.temperature_scale)
+  println(io, "pottemp_flux            = ", sfc.pottemp_flux)
+  println(io, "exchange_coeff_momentum = ", sfc.exchange_coeff_momentum)
+  println(io, "exchange_coeff_heat     = ", sfc.exchange_coeff_heat)
+  println(io, "-----------------------")
 end
 
 """
-    compute_MO_len(u, flux)
-
-Computes the Monin-Obukhov length (Eq. 3 Ref. Byun1990)
-"""
-compute_MO_len(u, flux) = - u^3 / (flux * k_Karman)
-
+    surface_conditions(u_ave::DT,
+                       θ_bar::DT,
+                       θ_ave::DT,
+                       θ_s::DT,
+                       Δz::DT,
+                       z_0_m::DT,
+                       z_0_h::DT,
+                       z::DT,
+                       F_m::DT,
+                       F_h::DT,
+                       a::DT,
+                       Pr::DT,
+                       pottemp_flux::DT
+                       ) where DT<:AbstractFloat
+
+Surface conditions given
+ - `u_ave` volume-averaged horizontal wind speed
+ - `θ_ave` volume-averaged potential temperature
+ - `θ_bar` basic potential temperature
+ - `θ_s` surface potential temperature
+ - `Δz` layer thickness (not spatial discretization)
+ - `z_0_m` roughness length for momentum
+ - `z_0_h` roughness length for heat
```

Comment:
Add a roughness length for moisture and other tracers?

Reply:
This will change things quite a bit. Which input parameters need their own version for its corresponding moisture/tracer variables? Also perhaps I should nix the case when `pottemp_flux` is given? (this is what was done in SCAMPy, but it's not what is done in the Nishizawa paper)

Reply:
We will need moisture fluxes (evaporation) and passive tracer fluxes (for the `n` passive tracers we'll have in the end, where `n` is a variable). It would be good for moisture to have a separate roughness length (by default set equal to heat), and for all other tracers to have one roughness length in common.